diff --git a/dist/index.js b/dist/index.js
index 7b395c44a699c114c26b337c27ac376f6d5ee313..73cc4de5a899420e2ae30c8f2a4da9a384da909c 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1433,7 +1433,9 @@ function prepareCallSettings({
   }
   return {
     maxTokens,
-    temperature: temperature != null ? temperature : 0,
+    ...temperature === undefined ? {} : {
+      temperature: temperature != null ? temperature : 0
+    },
     topP,
     topK,
     presencePenalty,
diff --git a/dist/index.mjs b/dist/index.mjs
index e396a3ac5e56ead453db0d46071dccbbebe9ce5d..c352650bd23970f2e2c36f95751cb8195fab5d63 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -1374,7 +1374,10 @@ function prepareCallSettings({
   }
   return {
     maxTokens,
-    temperature: temperature != null ? temperature : 0,
+    ...temperature === undefined ? {} : {
+      temperature: temperature != null ? temperature : 0
+    },
+    // temperature: temperature != null ? temperature : 0,
     topP,
     topK,
     presencePenalty,
